Put an array of all the color names in ColorUtils

marcelgerber opened the pull request:
So we (and extension developers) can reuse them whereever we want.
The actual regex is unchanged, it's just getting generated now.

Commit:
Put an array of all the color names in ColorUtils

le717 commented:
I like this method better than stuffing them in the regex, but I'll need to check what changes I need to make to the Inline Color Editor to use this. :+1:

le717 commented on src/utils/ColorUtils.js:
```diff
@@ -32,13 +32,21 @@ define(function (require, exports, module) {
+     * Sorted array of all the color names in the CSS Color Module Level 3 (http://www.w3.org/TR/css3-color/)
+     * @const @type {Array}
+     */
+    var COLOR_NAMES = ["aliceblue", "antiquewhite", "aqua", "aquamarine", "azure", "beige", "bisque", "black", "blanchedalmond", "blue", "blueviolet", "brown", "burlywood", "cadetblue", "chartreuse", "chocolate", "coral", "cornflowerblue", "cornsilk", "crimson", "cyan", "darkblue", "darkcyan", "darkgoldenrod", "darkgray", "darkgreen", "darkgrey", "darkkhaki", "darkmagenta", "darkolivegreen", "dar ...
```
Just in case this is accepted, can you go ahead and add `rebeccapurple` to this array (#10250)?

MarcelGerber commented:
They are still in the regex, so you don't have to change anything.

redmunds commented:
This looks good. I agree that `rebeccapurple` should be added to the list.

le717 commented:
@redmunds I will close #10252 in favor of this.

MarcelGerber commented:
I think we need #9596 first for rebeccapurple to work properly, right.
I don't know what happens right now if you try to open an color inline editor on rebeccapurple, though.

Commit:
Add rebeccapurple

MarcelGerber commented:
Added support for `rebeccapurple` (and changed comments in order to match the color names included).
Still, the Color Inline Editor doesn't work too well with `rebeccapurple` (see https://github.com/adobe/brackets/issues/10250#issuecomment-67748764) (Quick View works perfectly fine, btw), but imho we can still merge it as there won't be many out there wanting to edit `rebeccapurple` anyway, I guess.

Sooner or later, we should, of course, merge #9596 for full support.

le717 commented:
@MarcelGerber I am going to try to finish up #9596 today.

le717 commented:
@redmunds While you're online, can you look into merging this too so we can finish up `rebeccapurple` support?

le717 commented on src/utils/ColorUtils.js:
```diff
@@ -32,13 +32,23 @@ define(function (require, exports, module) {
     
     /**
+     * Sorted array of all the color names in the CSS Color Module Level 3 (http://www.w3.org/TR/css3-color/)
+     * and "rebeccapurple" from CSS Color Module Level 4
```
Minor nit, does not totally need fixing: I'd reword this to 

> Sorted array of all the color names in the CSS Color Module Level 3 and 4 (http://www.w3.org/TR/css3-color/).

I don't think we need to specifically state what color(s) from CSS4 is supported, as ideally we'd support all of them.

MarcelGerber replied:
I don't think Module Level 4 is ready yet, while Level 3 is.
While it's still in development, we should state which names are available.

redmunds commented:
Looks good. Merging.
